File: challenges/2026-09-21-harmonic-pair-counter/solution.java
import java.util.*;

public class solution {

    public static long countHarmonicPairs(List<Integer> freq, int k) {
        // TODO: implement
        return 0;
    }

    public static void main(String[] args) {
        List<Object[]> tests = new ArrayList<>();
        tests.add(new Object[]{"example 1", List.of(4, 6, 8, 10, 3), 2, 5L});
        tests.add(new Object[]{"example 2", List.of(2, 4, 8, 16), 4, 2L});
        tests.add(new Object[]{"example 3", List.of(5, 7, 11), 5, 0L});
        tests.add(new Object[]{"edge: k larger than every value", List.of(2, 4, 6), 100, 0L});
        tests.add(new Object[]{"edge: all elements equal to k", List.of(7, 7, 7), 7, 3L});

        for (Object[] test : tests) {
            String name = (String) test[0];
            @SuppressWarnings("unchecked")
            List<Integer> freq = (List<Integer>) test[1];
            int k = (Integer) test[2];
            long expected = (Long) test[3];
            long actual = countHarmonicPairs(freq, k);
            String status = actual == expected ? "PASS" : "FAIL";
            System.out.println("[" + status + "] " + name + ": expected=" + expected + " actual=" + actual);
        }
    }
}

File: challenges/2026-09-21-harmonic-pair-counter/solution.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public static long CountHarmonicPairs(List<int> freq, int k)
    {
        // TODO: implement
        return 0;
    }

    public static void Main(string[] args)
    {
        var tests = new List<(string Name, List<int> Freq, int K, long Expected)>
        {
            ("example 1", new List<int> { 4, 6, 8, 10, 3 }, 2, 5),
            ("example 2", new List<int> { 2, 4, 8, 16 }, 4, 2),
            ("example 3", new List<int> { 5, 7, 11 }, 5, 0),
            ("edge: k larger than every value", new List<int> { 2, 4, 6 }, 100, 0),
            ("edge: all elements equal to k", new List<int> { 7, 7, 7 }, 7, 3),
        };

        foreach (var (name, freq, k, expected) in tests)
        {
            var actual = CountHarmonicPairs(freq, k);
            string status = actual == expected ? "PASS" : "FAIL";
            Console.WriteLine($"[{status}] {name}: expected={expected} actual={actual}");
        }
    }
}
